VariantShredding.md

```diff
@@ -193,6 +192,7 @@ optional group event (VARIANT) {
 ```
 
 The group for each named field must use repetition level `required`.
+Readers may always assume the group is annotated correctly.
```

Comment:
I don't think this should be here. The annotation problem is temporary.

Reply:
Sorry. What do you mean temporary? 

From the arrow-go implementation (https://github.com/apache/arrow-go/pull/455), seems the readers may check the group and fail if it's not `required`.

Reply:
I don't think this is a requirement. The requirement that is already here is that writers produce `required` fields. Readers are free to accept data that is slightly malformed, but this is not a requirement. I don't think this should be in the spec.

Reply:
I removed this but in some places that we explicitly say "readers may assume ...." to set expectation and I think that is also not a hard requirement? Wondering if we do the same here so if it's `optional group event_type`, the readers may accept such data.

>> Readers may always assume that data is written correctly and that only `value` or `typed_value` is defined.